crates/op-consensus/src/receipt/envelope.rs

```diff
@@ -0,0 +1,196 @@
+use alloy::{
+    consensus::{Receipt, ReceiptWithBloom},
+    rpc::types::eth::Log,
+};
+use alloy_primitives::Bloom;
+
+/// Receipt envelope, as defined in [EIP-2718].
+///
+/// This enum distinguishes between tagged and untagged legacy receipts, as the
+/// in-protocol merkle tree may commit to EITHER 0-prefixed or raw. Therefore
+/// we must ensure that encoding returns the precise byte-array that was
+/// decoded, preserving the presence or absence of the `TransactionType` flag.
+///
+/// Transaction receipt payloads are specified in their respective EIPs.
+///
+/// [EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(tag = "type"))]
+#[non_exhaustive]
+pub enum ReceiptEnvelope<T = Log> {
+    /// Receipt envelope with no type flag.
+    #[cfg_attr(feature = "serde", serde(rename = "0x0", alias = "0x00"))]
+    Legacy(ReceiptWithBloom<T>),
+    /// Receipt envelope with type flag 1, containing a [EIP-2930] receipt.
+    ///
+    /// [EIP-2930]: https://eips.ethereum.org/EIPS/eip-2930
+    #[cfg_attr(feature = "serde", serde(rename = "0x1", alias = "0x01"))]
+    Eip2930(ReceiptWithBloom<T>),
+    /// Receipt envelope with type flag 2, containing a [EIP-1559] receipt.
+    ///
+    /// [EIP-1559]: https://eips.ethereum.org/EIPS/eip-1559
+    #[cfg_attr(feature = "serde", serde(rename = "0x2", alias = "0x02"))]
+    Eip1559(ReceiptWithBloom<T>),
+    /// Receipt envelope with type flag 2, containing a [EIP-4844] receipt.
+    ///
+    /// [EIP-4844]: https://eips.ethereum.org/EIPS/eip-4844
+    #[cfg_attr(feature = "serde", serde(rename = "0x3", alias = "0x03"))]
+    Eip4844(ReceiptWithBloom<T>),
+    /// Receipt envelope for Optimism's deposit transactions
+    #[cfg_attr(feature = "serde", serde(rename = "0x7E", alias = "0x7E"))]
+    Deposit(ReceiptWithBloom<T>),
+}
+
+impl<T> ReceiptEnvelope<T> {
+    /// Return the [`TxType`] of the inner receipt.
+    pub const fn tx_type(&self) -> TxType {
+        match self {
+            Self::Legacy(_) => TxType::Legacy,
+            Self::Eip2930(_) => TxType::Eip2930,
+            Self::Eip1559(_) => TxType::Eip1559,
+            Self::Eip4844(_) => TxType::Eip4844,
+        }
```

Comment:
isnt this missing the deposit tx? all of these methods should deref to the inner enum variant i think

Reply:
This is just a placeholder because consensus should be another pr i think

Reply:
k let's separate or block the PR on the other work then

Reply:
Removed the op-consensus crate from this pr completely